csharp/src/Microsoft.ML.OnnxRuntime/NativeMethods.shared.cs

```diff
@@ -248,6 +248,10 @@ public struct OrtApi
         public IntPtr InvokeOp;
         public IntPtr ReleaseOp;
         public IntPtr SessionOptionsAppendExecutionProvider;
+        public IntPtr CopyKernelInfo;
+        public IntPtr ReleaseKernelInfo;
+
+        public IntPtr GetTrainingApi;
```

Comment:
>GetTrainingApi [](http://example.com/codeflow?start=22&length=14)

Do we need this here? I could not find a delegate to invoke this. Also, one can not return a pointer to a C++ struct and use it in C#. I am seeing you created a separate native methods for training which is the right thing to do. However, this will need a comment bc it seems to be a dummy entry point.

Reply:
The delegate is here https://github.com/microsoft/onnxruntime/pull/12404/files#diff-12144653b66d58f87477d8db86ce277795d71b7142b91d29ee9e2ee46b46b185R42
The pointer to struct is done by https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.marshal.ptrtostructure?view=net-6.0